### PR TITLE
Dialogue clients do not fail when a Content-Length header is provided

### PR DESCRIPTION
**Pull request by carterkozak**
==COMMIT_MSG==
Dialogue clients do not fail when a Content-Lenght header is provided
==COMMIT_MSG==

### CODE REVIEW

**Comment by carterkozak** [comment]
Not sure this is ideal, PR for discussion.

**Comment by carterkozak** [comment]
Note that this would allow us to set the known content-length from the feign jaxrs shim as well, but we may want to add an OptionalLong to the RequestBody interface instead of pulling data from the request headers (same as we do for content-type) though that wouldn't be helpful for @jackwickham.

File: dialogue-target/src/main/java/com/palantir/dialogue/RequestBody.java [hunk 1]
```diff
@@ -31,6 +32,13 @@
     /** Returns <pre>true</pre> if {@link #writeTo(OutputStream)} may be invoked multiple times. */
     boolean repeatable();
 
+    /** Returns the {@code Content-Length} of the request body, if known. Note that there is no guarantee that
+     * this will result in a {@code Content-Length} request rather than {@code Transfer-Encoding: chunked}.
+     */
+    default OptionalLong length() {
+        return OptionalLong.empty();
+    }
```
Comment (iamdanfox) [inline dialogue-target/src/main/java/com/palantir/dialogue/RequestBody.java]: I thought we weren't intending to expose this in `:dialogue-target`, but just say if users happen to set a `Content-Length` header we might choose to respect it?

Comment (carterkozak) [reply dialogue-target/src/main/java/com/palantir/dialogue/RequestBody.java]: I could go either way on this — the changes to this class are in an isolated commit.
We definitely aren’t including any changes to BinaryRequestBody, conjure consumers don’t interact with this interface though.

Comment (ferozco) [reply dialogue-target/src/main/java/com/palantir/dialogue/RequestBody.java]: I think for now I'd prefer to go with respecting the `Content-Length` header if the user provides it rather that extending the Request interface

Comment (carterkozak) [reply dialogue-target/src/main/java/com/palantir/dialogue/RequestBody.java]: reverted that commit.

File: dialogue-apache-hc4-client/src/main/java/com/palantir/dialogue/hc4/ApacheHttpClientBlockingChannel.java [hunk 2]
```diff
@@ -215,7 +239,9 @@ public boolean isStreaming() {
         }
 
         @Override
-        public void consumeContent() {}
+        public void consumeContent() {
+            requestBody.close();
```
Comment (ferozco) [inline dialogue-apache-hc4-client/src/main/java/com/palantir/dialogue/hc4/ApacheHttpClientBlockingChannel.java:243]: cheeky little fix

**Review by ferozco — APPROVED** [review]

**Pull request merged**